bin/lint-migrations-file/src/lint_migrations_file.clj

```diff
@@ -36,70 +35,63 @@
   (let [ids (change-set-ids change-log)]
     (= ids (sort-by identity compare ids))))
 
-(defn- assert-no-types-in-change-set
-  "Walks over x (a changeset map) to ensure it doesn't add any columns of `target-types` (a set of strings).
-  `found-cols` is an atom of vector, in which any problematic changes to the `target-types` will be stored.
-
-  A partial application of this function will be passed to postwalk below.
-
-  TODO: add and conform to a spec instead?"
-  [target-types found-cols x]
-  {:pre [(set? target-types) (instance? clojure.lang.Atom found-cols)]}
-  (if
-    (map? x)
+(defn- check-change-use-types?
+  "Return `true` if change use any type in `types`."
+  [types change]
+  {:pre [(set? types)]}
+  (let [match-target-types? (fn [ttype]
+                              (contains? types (str/lower-case ttype)))]
     (cond
-      ;; a createTable or addColumn change; see if it adds a target-type col
-      (or (:createTable x) (:addColumn x))
-      (let [op     (cond (:createTable x) :createTable (:addColumn x) :addColumn)
-            cols   (filter (fn [col-def]
-                             (contains? target-types
-                                        (str/lower-case (or (get-in col-def [:column :type]) ""))))
-                     (get-in x [op :columns]))]
-        (doseq [col cols]
-          (swap! found-cols conj col))
-        x)
-
-      ;; a modifyDataType change; see if it changes a column to target-type
-      (:modifyDataType x)
-      (if (= target-types (str/lower-case (or (get-in x [:modifyDataType :newDataType]) "")))
-        (do (swap! found-cols conj x)
-            x)
-        x)
-
-      ;; some other kind of change; continue walking
-      :else x)
-    x))
+     ;; a createTable or addColumn change; see if it adds a target-type col
+     (or (:createTable change) (:addColumn change))
+     (let [op (cond (:createTable change) :createTable (:addColumn change) :addColumn)]
+       (some (fn [col-def]
+               (match-target-types? (get-in col-def [:column :type] "")))
+             (get-in change [op :columns])))
+
+     ;; a modifyDataType change; see if it change a column to target-type
+     (and (:modifyDataType change)
+          (match-target-types? (get-in change [:modifyDataType :newDataType] "")))
+     true)))
+
+(defn- check-change-set-use-types?
+  "Return true if `change-set` doesn't contain usage of any type in `types`."
+  [target-types change-set]
+  (some #(check-change-use-types? target-types %) (get-in change-set [:changeSet :changes])))
+
+(defn- check-change-log-use-types?
```

Comment:
IMO we've lost some readability here because this function doesn't have a docstring. `assert-no-types-in-change-set` had a pretty good explanation of what was happening.

This function is used in a few places and its implementation is complex enough to warrant a docstring. Something like: 
```
"Iterates over the change-set maps in the change-log and returns a list of problematic changes. A change is problematic if it adds/changes any columns to be of a type in `target-types `."
```

Reply:
Also, I noticed that everywhere this function is used, it is wrapped in a `not`. Perhaps the `not` can be added to the body of this function and it could be renamed to `assert-no-types-in-change-log`, similar to before.